misc/package/Data Files/MWSE/core/mcm/components/categories/Category.lua

```diff
@@ -66,24 +66,22 @@ function Category:update()
 end
 
 function Category:checkDisabled()
+	-- dont disable if there are no subcomponents
+	if next(self.components) == nil then return false end
+
 	-- If has variables and all are inGameOnly, disable Category
-	local isDisabled = true
-	local hasSettings = false
+	local allSubcomponentsDisabled = true
 	for _, component in ipairs(self.components) do
-		if component.componentType == "Setting" and component.variable then
-			hasSettings = true
-			if component.variable.inGameOnly == false then
-				isDisabled = false
-			end
-		elseif component.componentType == "Category" then
-			local componentDisabled = component:checkDisabled()
-			isDisabled = component:checkDisabled()
-			if componentDisabled then
-				hasSettings = true
+		-- only check nested categories if they're nonempty
+		if component.componentType == "Category" then
+			if next(component.components) ~= nil then
+				allSubcomponentsDisabled = allSubcomponentsDisabled and component:checkDisabled()
 			end
+		else
+			allSubcomponentsDisabled = allSubcomponentsDisabled and component:checkDisabled()
 		end
 	end
-	return (hasSettings and not tes3.player and isDisabled)
+	return (allSubcomponentsDisabled and not tes3.player)
```

Comment:
We can simplify this block a bit. Let's consider the following 3 conditions:
1. If a single nested component (Setting or a Category) isn't disabled, we don't want to disable the owning Category.
2. If we've checked that every single nested component of a category is disabled, we can disable a Category.
3. Let's consider empty Categories as not disablable.

Then, we can just iterate over the `self.components` and:
1. A subcomponent isn't disabled, `return false`
2. Make `return true` the default return.
3. `return false` in the above check.

Reply:
the approach you suggested here is much cleaner. i implemented it in the latest commit, as well as your earlier suggestion of using `table.empty` instead of `next(...) == nil`.